### ISSUE TITLE
Image validation fails on valid images

### ISSUE THREAD

**Issue by andrewhavens** [refile/refile#431]
Not sure which of these is the problem. But I'm finding that some valid images are not passing validation. I think it might be a mime type mismatch on the server, or maybe it's the inclusion of `charset=UTF8`. For example:

In my model:

``` ruby
attachment :attachment, type: :image
```
## Example 1:
- Filename: `.jpg`
- Mime type: `image/jpeg`
- Response header: `Content-Type: image/png;charset=UTF-8`
- URL: http://blogs.scientificamerican.com/blogs/cache/file/8197CEC8-990D-4905-AE984685E38CA6BF_agenda.jpg?w=600&h=335
## Example 2:
- Filename: `.png`
- Mime type: `image/jpeg`
- Response header: `Content-Type: image/jpeg; charset=utf-8`
- URL: https://www.asus.com/media/global/products/tlvlZTN8Y0VoKyiB/P_setting_fff_1_90_end_600.png

**Comment by Maxoos** [comment]
I've seen the same, not sure why. I don't think this rather excellent gem is maintained anymore or as often as needed.

**Comment by sobrinho** [comment]
I'm not seeing this issue by using `remote_{attachment}_url` or downloading it and uploading through the browser.

**Example 1:**

``` bash
λ curl -L 'http://blogs.scientificamerican.com/blogs/cache/file/8197CEC8-990D-4905-AE984685E38CA6BF_agenda.jpg?w=600&h=335' | md5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 25249  100 25249    0     0   163k      0 --:--:-- --:--:-- --:--:--  163k
a5a3fa6e967a6ee4606d00b7ca95fba8
```

```ruby
[15] pry(main)> x = User.create(remote_avatar_url: 'http://blogs.scientificamerican.com/blogs/cache/file/8197CEC8-990D-4905-AE984685E38CA6BF_agenda.jpg?w=600&h=335')
   (0.1ms)  begin transaction
  SQL (0.4ms)  INSERT INTO "users" ("avatar_id", "avatar_content_type", "created_at", "updated_at") VALUES (?, ?, ?, ?)  [["avatar_id", "eacebc48e86a002434bdbf9775892e6ae4f5d19a9d4f972ef46305498bd8"], ["avatar_content_type", "image/jpeg"], ["created_at", "2018-01-15 11:47:10.560940"], ["updated_at", "2018-01-15 11:47:10.560940"]]
   (0.8ms)  commit transaction
=> #<User:0x00007ffa3dbe03b8
 id: 10,
 name: nil,
 avatar_id: "eacebc48e86a002434bdbf9775892e6ae4f5d19a9d4f972ef46305498bd8",
 avatar_content_type: "image/jpeg",
 created_at: Mon, 15 Jan 2018 11:47:10 UTC +00:00,
 updated_at: Mon, 15 Jan 2018 11:47:10 UTC +00:00>
[16] pry(main)> Digest::MD5.hexdigest x.avatar.read
=> "a5a3fa6e967a6ee4606d00b7ca95fba8"
```

**Example 2:**

```bash
λ curl -L 'https://www.asus.com/media/global/products/tlvlZTN8Y0VoKyiB/P_setting_fff_1_90_end_600.png' | md5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 50825  100 50825    0     0   112k      0 --:--:-- --:--:-- --:--:--  112k
f8888c1ac70ea4da01846d3bf6ed4396
```

```ruby
[17] pry(main)> x = User.create(remote_avatar_url: 'https://www.asus.com/media/global/products/tlvlZTN8Y0VoKyiB/P_setting_fff_1_90_end_600.png')
   (0.1ms)  begin transaction
  SQL (0.3ms)  INSERT INTO "users" ("avatar_id", "avatar_content_type", "created_at", "updated_at") VALUES (?, ?, ?, ?)  [["avatar_id", "de0cb6cbf3de3bbbc3afb1a153ca0f596af711e9a67492774ecc5dd4800b"], ["avatar_content_type", "image/jpeg"], ["created_at", "2018-01-15 11:49:26.231822"], ["updated_at", "2018-01-15 11:49:26.231822"]]
   (0.9ms)  commit transaction
=> #<User:0x00007ffa3ab218a8
 id: 11,
 name: nil,
 avatar_id: "de0cb6cbf3de3bbbc3afb1a153ca0f596af711e9a67492774ecc5dd4800b",
 avatar_content_type: "image/jpeg",
 created_at: Mon, 15 Jan 2018 11:49:26 UTC +00:00,
 updated_at: Mon, 15 Jan 2018 11:49:26 UTC +00:00>
[18] pry(main)> Digest::MD5.hexdigest x.avatar.read
=> "f8888c1ac70ea4da01846d3bf6ed4396"
```

**Model:**

```ruby
class User < ApplicationRecord
  has_many :uploads, as: :uploadable, inverse_of: :uploadable, dependent: :destroy

  accepts_attachments_for :uploads, attachment: :file

  attachment :avatar, type: :image
end
```

But there is no UTF-8 on content-type response header anymore and I've seen some reports of this being a mistake causing trouble for IE: https://jira.atlassian.com/browse/JRASERVER-29545?src=confmacro

So, I think the validation is working as designed but I see the consequences of images out of our control being sent to to our servers.

Maybe we can fix this on the Refile side by "discarding" the encoding part of the response header.

PRs are welcome, if anyone is interested in doing this, please notify me here.

**Comment by sobrinho** [comment]
Closed by #569.
